driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/ListCollectionsPublisher.java

```diff
@@ -83,6 +84,26 @@ public interface ListCollectionsPublisher<TResult> extends Publisher<TResult> {
      */
     ListCollectionsPublisher<TResult> comment(@Nullable BsonValue comment);
 
+    /**
+     * Sets the timeoutMode for the cursor.
+     *
+     * <p>
+     *     Requires the {@code timeout} to be set, either in the {@link com.mongodb.MongoClientSettings},
+     *     via {@link MongoDatabase} or via {@link MongoCollection}
+     * </p>
+     * <p>
+     *     If the {@code timeout} is set then:
+     *     <ul>
+     *      <li>For non-tailable cursors, the default value of timeoutMode is {@link TimeoutMode#CURSOR_LIFETIME}</li>
+     *      <li>For tailable cursors, the default value of timeoutMode is {@link TimeoutMode#ITERATION} and its an error
```

Comment:
[optional] As far as I understand, only Find, ChangeStream, and some Aggregate operations could be tailable. If ListCollections operations cannot be tailable, I would suggest not mentioning 'tailable' in its Javadoc to avoid any confusion for users. What do you think?

I might be applied to other Publishers/Iterables.

Reply:
SGTM - Done